lib/src/editor.dart

```diff
@@ -241,11 +241,13 @@ class YamlEditor {
 
     if (path.isEmpty) {
       final start = _contents.span.start.offset;
-      final end = getContentSensitiveEnd(_contents);
+      var end = getContentSensitiveEnd(_contents);
       final lineEnding = getLineEnding(_yaml);
-      final edit = SourceEdit(
-          start, end - start, yamlEncodeBlock(valueNode, 0, lineEnding));
-
+      end = skipAndExtractCommentsInBlock(_yaml, end, null, lineEnding).$1;
```

Comment:
```suggestion
      final (endOffset, _) = skipAndExtractCommentsInBlock(_yaml, end, null, lineEnding);
      end = endOffset;
```
or similar, but let's avoid `.$1` if we can.


Or make `skipAndExtractCommentsInBlock` return a record with named properties. So we can do: `skipAndExtractCommentsInBlock(...).endOffset;`.